fdbclient/BlobStore.actor.cpp

```diff
@@ -150,7 +150,11 @@ Reference<BlobStoreEndpoint> BlobStoreEndpoint::fromString(std::string const &ur
 		StringRef prefix = t.eat("://");
 		if(prefix != LiteralStringRef("blobstore"))
 			throw format("Invalid blobstore URL prefix '%s'", prefix.toString().c_str());
-		StringRef cred = t.eat("@");
+
```

Comment:
`StringRef::eat()` supports discovering and returning empty strings.  The `if()` here is essentially checking to see if `eat()` _would_ return an empty string, and then avoiding calling it if it will.  

This seems less clear than simply keeping 
`StringRef cred = t.eat("@");`
 and instead of checking `if(cred.present())` later use `if(cred.size() == 0)`. 

 I think the latter reads more clearly.

Reply:
Disregard! I completely misread what this conditional is doing.